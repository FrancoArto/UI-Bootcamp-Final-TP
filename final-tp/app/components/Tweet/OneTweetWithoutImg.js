import React, { Component } from 'react';
import { View, TouchableOpacity, Image } from 'react-native';
import { Thumbnail, Text, Icon, Button } from 'native-base';
import styles from './oneTweet.style';
import { Font, AppLoading } from "expo";

class OneTweetWhitoutImg extends Component {

    constructor(props) {
        super(props);
        this.state = { loading: true };
    }

    async componentWillMount() {
        await Font.loadAsync({
            Roboto: require("native-base/Fonts/Roboto.ttf"),
            Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
        });
        this.setState({ loading: false });
    }

    render() {

        const { navigation } = this.props;

        if (this.state.loading) {
            return (
                <AppLoading />
            );
        } else {
            return (
                <View style={[styles.flexTweet]}>

                    <View style={[styles.flexHead]}>
                        <View style={[styles.flexUserImg]}>
                            <TouchableOpacity onPress={() => navigation.navigate('UserProfile', { user: navigation.getParam('user', 'USER_ERROR') })}>
                                <Thumbnail large source={{ uri: navigation.getParam('uri', 'URI_ERROR') }} />
                            </TouchableOpacity>
                        </View>

                        <View style={[styles.flexTitleColumn]}>
                            <View style={[styles.flexTitle]}>
                                <Text style={[styles.fontUserName]}> {navigation.getParam('user', 'USER_ERROR').name} </Text>
                            </View>

                            <View style={[styles.flexTitle]}>
                                <Text style={[styles.fontCountNameAndTime]}> @{navigation.getParam('countName', 'COUNT-USER_ERROR')}</Text>
                                <Text style={[styles.fontCountNameAndTime]}> - {navigation.getParam('timeAgo', 'TIME-AGO_ERROR')}</Text>
                            </View>
                        </View>
                    </View>

                    <View style={[styles.flexContent]}>
                        <Text style={[styles.fontMainContent]}> {navigation.getParam('mainContent', 'MAIN-CONTENT_ERROR')} </Text>
                    </View>

                    <View style={[styles.flexIcons]}>
                        <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                            <Icon type="SimpleLineIcons" name="heart" style={[styles.sizeIcons]} />
                            <Text style={[styles.fontNumberIcons]}>{this.props.favorite_count}</Text>
                        </Button>

                        <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                            <Icon type="SimpleLineIcons" name="refresh" style={[styles.sizeIcons]} />
                            <Text style={[styles.fontNumberIcons]}>{this.props.retweet_count}</Text>
                        </Button>

                        <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                            <Icon type="SimpleLineIcons" name="bubble" style={[styles.sizeIcons]} />
                            <Text style={[styles.fontNumberIcons]}></Text>
                        </Button>
                    </View>
                </View>
            );
        }
    }
}
export default OneTweetWhitoutImg;
