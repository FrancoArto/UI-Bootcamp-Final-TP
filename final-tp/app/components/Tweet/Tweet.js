import React, { PureComponent } from 'react';
import { View, TouchableOpacity, Image } from 'react-native';
import { Thumbnail, Text, Icon, Button } from 'native-base';
import styles from './tweet.style';
import { Font, AppLoading } from "expo";

import TimeAgo from 'javascript-time-ago';
import en from 'javascript-time-ago/locale/en';
import moment from 'moment-timezone';
TimeAgo.addLocale(en);

class Tweet extends PureComponent {

    constructor(props) {
        super(props);
        this.state = { loading: true };
        _isMounted = false


        this.goToUserProfile = this.goToUserProfile.bind(this)
        this.handleOnPress = this.handleOnPress.bind(this)

    }

    toTimeZone(apiHour) {
        const timeAgo = new TimeAgo('en-EN');
        let splitCompleteHour = apiHour.split(" ", apiHour.lenght);

        let momentFormat = moment(splitCompleteHour[1] + " " + splitCompleteHour[2] + ", " + splitCompleteHour[5] + " " + splitCompleteHour[3], "MMM ddd YYYY HH:mm:ss")

        let EN = momentFormat.tz('Europe/London')
        let ARG = EN.clone().tz('America/Argentina/Buenos_Aires');

        let parseFormat = moment(ARG).format('HH mm ss');
        let splitARGhour = parseFormat.split(" ", parseFormat.length);

        return timeAgo.format(Date.now() - splitARGhour[0] * splitARGhour[1] * splitARGhour[2] * 1000, 'twitter')
    }

    navigationProp() {
        return this.props.navigationProp
    }

    async componentWillMount() {
        await Font.loadAsync({
            Roboto: require("native-base/Fonts/Roboto.ttf"),
            Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
        });
        if (this._isMounted) {
            this.setState({ loading: false });
        }
    }
    componentDidMount() {
        this._isMounted = true;
    }
    componentWillUnmount() {
        this._isMounted = false;
    }

    goToUserProfile() {
        this.props.goToUserProfile(this.props.user.id)
    }

    handleOnPress() {
        this.props.onTweetPress(this.props.id_str)
    }

    render() {
        if (this.state.loading) {
            return (
                <AppLoading />
            );

        } else {

            return (
                <TouchableOpacity onPress={this.handleOnPress} >

                    <View style={[styles.flexTweet]}>

                        <TouchableOpacity onPress={this.goToUserProfile}>
                            <View style={[styles.flexUserImg]}>
                                <Thumbnail small source={{ uri: this.props.uri }} />
                            </View>
                        </TouchableOpacity>

                        <View style={[styles.flexRightSide]}>

                            <View style={[styles.flexTitle]}>
                                <Text style={[styles.fontUserName]}> {this.props.user.name} </Text>
                                <Text style={[styles.fontCountNameAndTime]}> @{this.props.user.screen_name} - {this.props.timeAgo} </Text>
                            </View>

                            <View style={[styles.flexContent]}>
                                <Text style={[styles.fontMainContent]}> {this.props.mainContent} </Text>
                            </View>

                            {this.props.media &&
                                <View style={[styles.flexImage, styles.extraMarginWhitImg]}>
                                    <Image small source={{
                                        uri: this.props.media[0].media_url_https
                                    }}
                                        style={[styles.imageTweet]} />
                                </View>
                            }


                            <View style={[styles.flexIcons]}>

                                <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                                    <Icon type="SimpleLineIcons" name="heart" style={[styles.sizeIcons]} />
                                    <Text style={[styles.fontNumberIcons]}>{this.props.favorite_count}</Text>
                                </Button>

                                <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                                    <Icon type="SimpleLineIcons" name="refresh" style={[styles.sizeIcons]} />
                                    <Text style={[styles.fontNumberIcons]}>{this.props.retweet_count}</Text>
                                </Button>

                                <Button iconLeft transparent primary style={[styles.buttonIcons]}>
                                    <Icon type="SimpleLineIcons" name="bubble" style={[styles.sizeIcons]} />
                                    <Text style={[styles.fontNumberIcons]}></Text>
                                </Button>
                            </View>

                        </View>
                    </View>
                </TouchableOpacity>
            );
        }
    }
}

export default Tweet;
