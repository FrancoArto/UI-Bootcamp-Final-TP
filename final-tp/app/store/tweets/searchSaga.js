import { GET_SEARCHTWEETS_URL, GET_APPENDRESULTS_URL } from '../../api/apiUrls';
import { takeLatest, put, call, select, all } from 'redux-saga/effects';
import { FETCH_SEARCH_BEGIN, fetchSearchSuccess, fetchSearchError, FETCH_MORE_RESULTS_REQUEST, fetchMoreResultsSuccess, fetchMoreResultsFailure } from './searchTweetsActions';

const getSearchText = state => state.searchTweetsReducer.searchText
const getMaxId = state => state.searchTweetsReducer.data[state.searchTweetsReducer.data.length - 1].id
const count = 50


function* fetchSearchResults() {  
  try {
    const searchText = yield select(getSearchText)
    const response = yield call(fetch, GET_SEARCHTWEETS_URL(searchText))
    const data = yield call([response, "json"]);
    yield put(fetchSearchSuccess(data.statuses))
  } catch (er) {
    yield put(fetchSearchError(er))
  }
}

function* searchTweets() {
  yield takeLatest(FETCH_SEARCH_BEGIN, fetchSearchResults)   
}


function* fetchResultsToAppend() {
  try {
    const searchText = yield select(getSearchText)
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDRESULTS_URL(searchText, count, maxId))
    const results = yield call([response, "json"]);
    const resultArray = results.statuses
    const data = resultArray.slice(1)
    yield put(fetchMoreResultsSuccess(data))
  } catch (er) {
    yield put(fetchMoreResultsFailure(er))
  }
}

function* appendResults() {
  yield takeLatest(FETCH_MORE_RESULTS_REQUEST, fetchResultsToAppend)   
}



export function* searchSaga() {
  yield all([
    searchTweets(),
    appendResults()
  ])
}