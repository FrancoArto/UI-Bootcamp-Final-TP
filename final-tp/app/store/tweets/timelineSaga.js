import { GET_TIMELINE_URL, GET_APPENDTWEETS_URL } from '../../api/apiUrls';
import { FETCH_TIMELINE_BEGIN, fetchTimelineSuccess, fetchTimelineFailure, FETCH_MORE_TWEETS_REQUEST, fetchMoreTweetsSuccess, fetchMoreTweetsFailure } from './timeLineActions'
import { takeLatest, put, call, all, select } from 'redux-saga/effects';

const twitTimelineCount = 50;

function* fetchTimeline() {
  try {
    const response = yield call(fetch, GET_TIMELINE_URL(twitTimelineCount))
    const data = yield call([response, "json"]);
    yield put(fetchTimelineSuccess(data))
  } catch (er) {
    yield put(fetchTimelineFailure(er))
  }
}

function* loadTimeline() {
  yield takeLatest(FETCH_TIMELINE_BEGIN, fetchTimeline)   
}

const getMaxId = state => state.timeLineReducer.data[state.timeLineReducer.data.length - 1].id


function* fetchTweetsToAppend() {
  try {
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDTWEETS_URL(twitTimelineCount, maxId))
    const array = yield call([response, "json"]);
    const data = array.slice(1)
    yield put(fetchMoreTweetsSuccess(data))
  } catch (er) {
    yield put(fetchMoreTweetsFailure(er))
  }
}

function* appendTweets() {
  yield takeLatest(FETCH_MORE_TWEETS_REQUEST, fetchTweetsToAppend)   
}

export function* timelineSaga() {
  yield all([
    loadTimeline(),
    appendTweets()
  ])
}