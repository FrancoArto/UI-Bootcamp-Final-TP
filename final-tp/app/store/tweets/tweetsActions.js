export const FETCH_SEARCH_BEGIN = 'FETCH_SEARCH_BEGIN'
export const FETCH_SEARCH_SUCCESS = 'FETCH_SEARCH_SUCCESS'
export const FETCH_SEARCH_FAILURE = 'FETCH_SEARCH_FAILURE'
export const FETCH_MORE_RESULTS_REQUEST = 'FETCH_MORE_RESULTS_REQUEST'
export const FETCH_MORE_RESULTS_SUCCESS = 'FETCH_MORE_RESULTS_SUCCESS'
export const FETCH_MORE_RESULTS_FAILURE = 'FETCH_MORE_RESULTS_FAILURE'
export const FETCH_TIMELINE_BEGIN = 'FETCH_TIMELINE_BEGIN'
export const FETCH_TIMELINE_SUCCESS = 'FETCH_TIMELINE_SUCCESS'
export const FETCH_TIMELINE_FAILURE = 'FETCH_TIMELINE_FAILURE'
export const FETCH_MORE_TWEETS_REQUEST = 'FETCH_MORE_TWEETS_REQUEST'
export const FETCH_MORE_TWEETS_SUCCESS = 'FETCH_MORE_TWEETS_SUCCESS'
export const FETCH_MORE_TWEETS_FAILURE = 'FETCH_MORE_TWEETS_FAILURE'
export const FETCH_USERTIMELINE_BEGIN = 'FETCH_USERTIMELINE_BEGIN'
export const FETCH_USERTIMELINE_SUCCESS = 'FETCH_USERTIMELINE_SUCCESS'
export const FETCH_USERTIMELINE_FAILURE = 'FETCH_USERTIMELINE_FAILURE'
export const FETCH_SINGLETWEET_BEGIN = 'FETCH_SINGLETWEET_BEGIN'
export const FETCH_SINGLETWEET_SUCCESS = 'FETCH_SINGLETWEET_SUCCESS'
export const FETCH_SINGLETWEET_FAILURE = 'FETCH_SINGLETWEET_FAILURE'


export const fetchTweetsTimeline = () => ({
    type: FETCH_TIMELINE_BEGIN
})

export const fetchTimelineSuccess = (data) => ({
    type: FETCH_TIMELINE_SUCCESS,
    payload: data
})

export const fetchTimelineFailure = (error) => ({
    type: FETCH_TIMELINE_FAILURE,
    payload: error
})

export const fetchMoreTweets = () => ({
    type: FETCH_MORE_TWEETS_REQUEST,
})

export const fetchMoreTweetsSuccess = (data) => ({
    type: FETCH_MORE_TWEETS_SUCCESS,
    payload: data
})

export const fetchMoreTweetsFailure = (error) => ({
    type: FETCH_MORE_TWEETS_FAILURE,
    payload: error
})

export const fetchSearchBegin = (searchText) => ({
    type: FETCH_SEARCH_BEGIN,
    payload: searchText
});

export const fetchSearchSuccess = (twitsArray) => ({
    type: FETCH_SEARCH_SUCCESS,
    payload: twitsArray
});

export const fetchSearchError = error => ({
    type: FETCH_SEARCH_FAILURE,
    payload: error
});

export const fetchMoreResults = () => ({
    type: FETCH_MORE_RESULTS_REQUEST,
})

export const fetchMoreResultsSuccess = (data) => ({
    type: FETCH_MORE_RESULTS_SUCCESS,
    payload: data
})

export const fetchMoreResultsFailure = (error) => ({
    type: FETCH_MORE_RESULTS_FAILURE,
    payload: error
})

export const fetchUserTimelineBegin = () => ({
    type: FETCH_USERTIMELINE_BEGIN,
});

export const fetchUserTimelineSuccess = (twitsArray) => ({
    type: FETCH_USERTIMELINE_SUCCESS,
    payload: twitsArray
});

export const fetchUserTimelineError = error => ({
    type: FETCH_USERTIMELINE_FAILURE,
    payload: error
});

export const fetchSingleTweetBegin = (tweetId) => ({
    type: FETCH_SINGLETWEET_BEGIN,
    payload: tweetId
});

export const fetchSingleTweetSuccess = (tweet) => ({
    type: FETCH_SINGLETWEET_SUCCESS,
    payload: tweet
});

export const fetchSingleTweetError = error => ({
    type: FETCH_SINGLETWEET_FAILURE,
    payload: error
});