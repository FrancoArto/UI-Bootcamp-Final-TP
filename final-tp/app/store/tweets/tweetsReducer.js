import {
  FETCH_SEARCH_BEGIN,
  FETCH_SEARCH_SUCCESS,
  FETCH_SEARCH_FAILURE,
  FETCH_MORE_RESULTS_REQUEST,
  FETCH_MORE_RESULTS_SUCCESS,
  FETCH_MORE_RESULTS_FAILURE,
  FETCH_TIMELINE_BEGIN,
  FETCH_TIMELINE_SUCCESS,
  FETCH_TIMELINE_FAILURE,
  FETCH_MORE_TWEETS_REQUEST,
  FETCH_MORE_TWEETS_SUCCESS,
  FETCH_MORE_TWEETS_FAILURE,
  FETCH_USERTIMELINE_BEGIN,
  FETCH_USERTIMELINE_SUCCESS,
  FETCH_USERTIMELINE_FAILURE,
  FETCH_SINGLETWEET_BEGIN,
  FETCH_SINGLETWEET_SUCCESS,
  FETCH_SINGLETWEET_FAILURE
} from './tweetsActions'
import { tweetsInitialState } from '../initialState'


const tweetsReducer = (state = tweetsInitialState, action) => {
  switch (action.type) {
    case FETCH_SEARCH_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
        searchText: action.payload
      }
    }
    case FETCH_SEARCH_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        searchResults: action.payload
      }
    }
    case FETCH_SEARCH_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        searchText: null,
        searchResults: []
      }
    }
    case FETCH_MORE_RESULTS_REQUEST: {
      return {
        ...state,
        loading: false,
        error: null,
      }
    }
    case FETCH_MORE_RESULTS_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        searchResults: state.searchResults.concat(...action.payload),
      }
    }
    case FETCH_MORE_RESULTS_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        searchResults: []
      }
    }
    case FETCH_TIMELINE_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
      }
    }
    case FETCH_TIMELINE_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        data: action.payload,
      }
    }
    case FETCH_TIMELINE_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        data: [],
      }
    }
    case FETCH_MORE_TWEETS_REQUEST: {
      return {
        ...state,
        loading: false,
        error: null
      }
    }
    case FETCH_MORE_TWEETS_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        data: state.data.concat(...action.payload),
      }
    }
    case FETCH_MORE_TWEETS_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        data: [],
      }
    }
    case FETCH_USERTIMELINE_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
      }
    }
    case FETCH_USERTIMELINE_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        userTimeline: action.payload,
      }
    }
    case FETCH_USERTIMELINE_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        userTimeline: [],
      }
    }
    case FETCH_SINGLETWEET_BEGIN: {
      return {
        ...state,
        loadingSingleTweet: true,
        error: null,
        singleTweetId: action.payload
      }
    }
    case FETCH_SINGLETWEET_SUCCESS: {
      return {
        ...state,
        loadingSingleTweet: false,
        error: null,
        singleTweet: action.payload,
      }
    }
    case FETCH_SINGLETWEET_FAILURE: {
      return {
        ...state,
        loadingSingleTweet: false,
        error: action.payload,
        singleTweet: {},
        singleTweetId: null
      }
    }
    default:
      return state;
  }
};
export default tweetsReducer;