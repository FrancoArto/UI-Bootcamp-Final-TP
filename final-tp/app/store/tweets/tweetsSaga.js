import { GET_SEARCHTWEETS_URL, GET_APPENDRESULTS_URL, GET_TIMELINE_URL, GET_APPENDTWEETS_URL, GET_USERTIMELINE_URL, GET_SINGLETWEET_URL } from '../../api/apiUrls';
import { takeLatest, put, call, select, all } from 'redux-saga/effects';
import { FETCH_SEARCH_BEGIN, 
  fetchSearchSuccess, 
  fetchSearchError, 
  FETCH_MORE_RESULTS_REQUEST, 
  fetchMoreResultsSuccess, 
  fetchMoreResultsFailure,
  FETCH_TIMELINE_BEGIN, 
  fetchTimelineSuccess, 
  fetchTimelineFailure, 
  FETCH_MORE_TWEETS_REQUEST, 
  fetchMoreTweetsSuccess, 
  fetchMoreTweetsFailure,
  fetchUserTimelineSuccess,
  fetchUserTimelineError,
  FETCH_USERTIMELINE_BEGIN,
  fetchUserTimelineBegin,
  FETCH_SINGLETWEET_BEGIN,
  fetchSingleTweetSuccess,
  fetchSingleTweetError
} from './tweetsActions';
import { getMaxId, getSearchText, getSingleTweetId } from './tweetsSelector';
import { FETCH_USERDATA_SUCCESS } from '../users/userActions';
import { getUser } from '../users/userSelector';

const count = 50


export function* fetchSearchResults() {  
  try {
    const searchText = yield select(getSearchText)
    const response = yield call(fetch, GET_SEARCHTWEETS_URL(searchText))
    const data = yield call([response, "json"]);
    yield put(fetchSearchSuccess(data.statuses))
  } catch (er) {
    yield put(fetchSearchError(er))
  }
}

function* searchTweets() {
  yield takeLatest(FETCH_SEARCH_BEGIN, fetchSearchResults)   
}


export function* fetchResultsToAppend() {
  try {
    const searchText = yield select(getSearchText)
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDRESULTS_URL(searchText, count, maxId))
    const results = yield call([response, "json"]);
    yield put(fetchMoreResultsSuccess(results.statuses.slice(1)))
  } catch (er) {
    yield put(fetchMoreResultsFailure(er))
  }
}

function* appendResults() {
  yield takeLatest(FETCH_MORE_RESULTS_REQUEST, fetchResultsToAppend)   
}




export function* fetchTimeline() {
  try {
    const response = yield call(fetch, GET_TIMELINE_URL(count))
    const data = yield call([response, "json"]);
    yield put(fetchTimelineSuccess(data))
  } catch (er) {
    yield put(fetchTimelineFailure(er))
  }
}

function* loadTimeline() {
  yield takeLatest(FETCH_TIMELINE_BEGIN, fetchTimeline)   
}



export function* fetchTweetsToAppend() {
  try {
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDTWEETS_URL(count, maxId))
    const array = yield call([response, "json"]);
    yield put(fetchMoreTweetsSuccess(array.slice(1)))
  } catch (er) {
    yield put(fetchMoreTweetsFailure(er))
  }
}

function* appendTweets() {
  yield takeLatest(FETCH_MORE_TWEETS_REQUEST, fetchTweetsToAppend)   
}

export function* fetchUserTimeline() {
  try {
    const user = yield select(getUser)
    const response = yield call(fetch, GET_USERTIMELINE_URL(user, count))
    const data = yield call([response, "json"]);
    yield put(fetchUserTimelineSuccess(data))
  } catch (er) {
    yield put(fetchUserTimelineError(er))
  }
}

function* userDataSucceeded() {
  yield takeLatest(FETCH_USERDATA_SUCCESS, requestUserTimelineFetch)
}

export function* requestUserTimelineFetch() {
  yield put(fetchUserTimelineBegin())
}


function* userTimeline() {
  yield takeLatest(FETCH_USERTIMELINE_BEGIN, fetchUserTimeline)
}

export function* fetchSingleTweet() {
  try {
    const tweetId = yield select(getSingleTweetId)
    const response = yield call(fetch, GET_SINGLETWEET_URL(tweetId))
    const data = yield call([response, "json"]);
    yield put(fetchSingleTweetSuccess(data))
  } catch (er) {
    yield put(fetchSingleTweetError(er))
  }
}

function* singleTweet() {
  yield takeLatest(FETCH_SINGLETWEET_BEGIN, fetchSingleTweet)
}


export function* tweetsSaga() {
  yield all([
    loadTimeline(),
    appendTweets(),
    searchTweets(),
    appendResults(),
    userDataSucceeded(),
    userTimeline(),
    singleTweet()
  ])
}