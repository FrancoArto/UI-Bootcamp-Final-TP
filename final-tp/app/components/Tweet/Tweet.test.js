import React from 'react';
import { mount, shallow } from 'enzyme'
import { singleTweetData } from '../../store/tweets/tweetsMock';
import Tweet from './Tweet';

const item = singleTweetData

it('renders without crashing', () => {
  const rendered = shallow(
    <Tweet
      id_str={item.id_str}
      user={item.user}
      mainContent={item.text}
      uri={item.user.profile_image_url_https}
      favorite_count={item.favorite_count}
      retweet_count={item.retweet_count}
      media={item.entities.media}
      created_at={item.created_at}
      media={item.entities.media}
    />)
  expect(rendered).toMatchSnapshot();
});

it('renders loaded tweet', () => {
  const rendered = shallow(
    <Tweet
      id_str={item.id_str}
      user={item.user}
      mainContent={item.text}
      uri={item.user.profile_image_url_https}
      favorite_count={item.favorite_count}
      retweet_count={item.retweet_count}
      media={item.entities.media}
      created_at={item.created_at}
      media={item.entities.media}
    />)
  rendered.setState({
    loading: false
  })
  rendered.update()
  expect(rendered).toMatchSnapshot();
});

it('handles tweet press', () => {
  const onTweetPress = jest.fn()
  const wrapper = mount(<Tweet
    id_str={item.id_str}
    user={item.user}
    mainContent={item.text}
    uri={item.user.profile_image_url_https}
    favorite_count={item.favorite_count}
    retweet_count={item.retweet_count}
    media={item.entities.media}
    created_at={item.created_at}
    media={item.entities.media}
    onTweetPress={onTweetPress} />)

  wrapper.setState({
    loading: false
  })
  wrapper.update()
  const tweet = wrapper.find('TouchableOpacity').first();
  tweet.props().onPress();
  expect(onTweetPress).toHaveBeenCalled()
});

it('handles user profile link', () => {
  const goToUserProfile = jest.fn()
  const wrapper = mount(<Tweet
    id_str={item.id_str}
    user={item.user}
    mainContent={item.text}
    uri={item.user.profile_image_url_https}
    favorite_count={item.favorite_count}
    retweet_count={item.retweet_count}
    media={item.entities.media}
    created_at={item.created_at}
    media={item.entities.media}
    goToUserProfile={goToUserProfile} />)

  wrapper.setState({
    loading: false
  })
  wrapper.update()
  const tweet = wrapper.find('TouchableOpacity').at(1);
  tweet.props().onPress();
  expect(goToUserProfile).toHaveBeenCalled()
});