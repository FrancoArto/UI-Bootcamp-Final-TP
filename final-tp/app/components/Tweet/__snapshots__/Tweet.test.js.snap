// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loaded tweet 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#D5D5D5",
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Component
        style={
          Array [
            Object {
              "flex": 0.9,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "padding": 12.5,
            },
          ]
        }
      >
        <Styled(Thumbnail)
          small={true}
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            }
          }
        />
      </Component>
    </TouchableOpacity>
    <Component
      style={
        Array [
          Object {
            "flex": 6.8,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginLeft": 0,
            "paddingLeft": 0,
            "paddingRight": 18.75,
          },
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingBottom": 12.5,
              "paddingTop": 12.5,
            },
          ]
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#040404",
                "fontSize": 31.25,
                "fontWeight": "bold",
              },
            ]
          }
        >
           
          Clarín
           
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#D5D5D5",
                "fontSize": 31.25,
              },
            ]
          }
        >
           @
          clarincom
           - 
           
        </Styled(Text)>
      </Component>
      <Component
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#5D5D5D",
                "fontSize": 26.785714285714285,
              },
            ]
          }
        >
           
          Detuvieron a un cura en San Pedro: lo acusan de abusar de nenes de un jardín de infantes https://t.co/Cqm0vehy5P https://t.co/OLSi1XswSf
           
        </Styled(Text)>
      </Component>
      <Component
        style={
          Array [
            Object {
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingBottom": 9.375,
              "paddingTop": 9.375,
            },
            Object {
              "marginRight": 50,
            },
          ]
        }
      >
        <Image
          small={true}
          source={
            Object {
              "uri": "https://pbs.twimg.com/media/DtqVRn_XgAAVAHD.jpg",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 20,
                "flex": 1,
                "flexDirection": "row",
                "height": 250,
              },
            ]
          }
        />
      </Component>
      <Component
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginLeft": 0,
              "marginRight": 15,
              "paddingBottom": 9.375,
              "paddingLeft": 0,
              "paddingTop": 9.375,
            },
          ]
        }
      >
        <Styled(Button)
          iconLeft={true}
          primary={true}
          style={
            Array [
              Object {
                "margin": 0,
                "padding": 0,
              },
            ]
          }
          transparent={true}
        >
          <Styled(Icon)
            name="heart"
            style={
              Array [
                Object {
                  "color": "#909090",
                  "fontSize": 37.5,
                  "marginLeft": 0,
                  "paddingLeft": 0,
                },
              ]
            }
            type="SimpleLineIcons"
          />
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "#5D5D5D",
                  "fontSize": 26.785714285714285,
                  "marginRight": 16.666666666666668,
                },
              ]
            }
          >
            26
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          iconLeft={true}
          primary={true}
          style={
            Array [
              Object {
                "margin": 0,
                "padding": 0,
              },
            ]
          }
          transparent={true}
        >
          <Styled(Icon)
            name="refresh"
            style={
              Array [
                Object {
                  "color": "#909090",
                  "fontSize": 37.5,
                  "marginLeft": 0,
                  "paddingLeft": 0,
                },
              ]
            }
            type="SimpleLineIcons"
          />
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "#5D5D5D",
                  "fontSize": 26.785714285714285,
                  "marginRight": 16.666666666666668,
                },
              ]
            }
          >
            14
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          iconLeft={true}
          primary={true}
          style={
            Array [
              Object {
                "margin": 0,
                "padding": 0,
              },
            ]
          }
          transparent={true}
        >
          <Styled(Icon)
            name="bubble"
            style={
              Array [
                Object {
                  "color": "#909090",
                  "fontSize": 37.5,
                  "marginLeft": 0,
                  "paddingLeft": 0,
                },
              ]
            }
            type="SimpleLineIcons"
          />
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "#5D5D5D",
                  "fontSize": 26.785714285714285,
                  "marginRight": 16.666666666666668,
                },
              ]
            }
          />
        </Styled(Button)>
      </Component>
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`renders without crashing 1`] = `<AppLoading />`;
